src/plots/cartesian/axes.js

```diff
@@ -2182,12 +2222,19 @@ axes.doTicksSingle = function(gd, arg, skipTitle) {
                     break;
                 }
             }
-            var rng = Lib.simpleMap(ax.range, ax.r2l),
-                showZl = (rng[0] * rng[1] <= 0) && ax.zeroline &&
+            var rng = Lib.simpleMap(ax.range, ax.r2l);
+            var zlData = {x: 0, id: axid};
```

Comment:
No need this do in this PR (opening a new issue would be fine), but we'll need to propagate these changes to large-sploms' regl-based grid line logic:

https://github.com/plotly/plotly.js/blob/b5f8b2300d041f12df90647fe61dba93f7d10c4b/src/traces/splom/base_plot.js#L171-L183

where I didn't just reuse the `axes.js` logic because I wanted to avoid looping over the traces (to compute that `hasBarsOrFill` boolean). Thinking about this again, looping over traces isn't really a performance hit for sploms (I can't really imagine a user wanting to plot more 10 splom traces on the same graph).

So it would be worth trying to expose and :recycle: this logic here for large-sploms.

Reply:
Ah, thanks for pointing that out! https://github.com/plotly/plotly.js/issues/2938